src/Compilers/VisualBasic/Portable/Scanner/CharacterInfo.vb

```diff
@@ -257,7 +257,19 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
 
         Friend Shared Function BeginsBaseLiteral(c As Char) As Boolean
             Return (c = "H"c Or c = "O"c Or c = "B"c Or c = "h"c Or c = "o"c Or c = "b"c) OrElse
-                    (IsFullWidth(c) AndAlso (c = FULLWIDTH_LATIN_CAPITAL_LETTER_H Or c = FULLWIDTH_LATIN_CAPITAL_LETTER_O Or c = FULLWIDTH_LATIN_CAPITAL_LETTER_B Or c = FULLWIDTH_LATIN_SMALL_LETTER_H Or c = FULLWIDTH_LATIN_SMALL_LETTER_O Or c = FULLWIDTH_LATIN_CAPITAL_LETTER_B))
+                    (IsFullWidth(c) AndAlso (BeginsBaseLiteral_Hex(c) OrElse BeginsBaseLiteral_Oct(c) OrElse BeginsBaseLiteral_Bin(c)))
+        End Function
+
+        Private Shared Function BeginsBaseLiteral_Hex(c As Char) As Boolean
```

Comment:
The names of the methods is a little misleading because they only handle full width characters. Perhaps inline the methods.

Reply:
Alternatively, use a `Select Case c` statement in `BeginsBaseLiteral` and remove the `IsFullWidth(c)` check.

Reply:
@cston I've reworked the method names.